spring-integration-jpa/src/main/java/org/springframework/integration/jpa/support/parametersource/ExpressionEvaluatingParameterSourceFactory.java

```diff
@@ -139,23 +139,21 @@ public Object getValue(String paramName) {
 					});
 		}
 
-		@Nullable
-		private Object obtainParameterValue(JpaParameter jpaParameter) {
-			Object value = null;
-			if (jpaParameter.getValue() != null) {
-				value = jpaParameter.getValue();
-			}
-			if (jpaParameter.getExpression() != null) {
+		private @Nullable Object obtainParameterValue(JpaParameter jpaParameter) {
+			Object value = jpaParameter.getValue();
+			if (value == null && jpaParameter.getExpression() != null) {
```

Comment:
I agree with change.    Should we add this as a note, since it is a change in behavior?

Reply:
I don’t think so. 
I tried to keep outcome of the method the same. Semantically logic is not changed. Just perspective how the value is handled .
What do you think I have broken since it gives an impression that behavior has changed?

Reply:
So, the contract is this:
```
	/**
	 * Instantiates a new Jpa Parameter without a name. This is useful for specifying
	 * positional Jpa parameters.
	 * @param value If null, the expression property must be set
	 * @param expression If null, the value property must be set
	 */
	public JpaParameter(@Nullable Object value, @Nullable String expression) {
```
Therefore an existing logic is wrong: we really should not go and check for `getExression()` if `getValue()` is not null.
It might rely on assumption that expression is not set, but I don't think we can rely on that.
With that I think it is better to honor that contract even more.